tests/test_threadpoolctl.py

```diff
@@ -587,3 +587,34 @@ def test_openblas_threading_layer():
         pytest.skip("Unknown OpenBLAS threading layer.")
 
     assert threading_layer in expected_openblas_threading_layers
+
+
+def test_threadpool_controller_as_decorator():
+    # Check that using the decorator can be nested and is restricted to the scope of
+    # the decorated function.
+    controller = ThreadpoolController()
+    original_info = controller.info()
+
+    if not controller.select(user_api="blas"):
+        pytest.skip(f"Requires a blas runtime.")
```

Comment:
Should we also skip if there is less than 2 cpus like in the nested case?
```suggestion
    blas_ctls = controller.select(user_api="blas")
    if not blas_ctls or any(ctl.num_threads < 2 for ctl in blas_ctls):
        pytest.skip(f"Requires a blas runtime and more than 2 CPUs.")
```

Reply:
yep makes sense. Done (I split the 2 skipping options to know exactly why it was skipped)